### ISSUE TITLE
Not receiving notifications after having no internet connection

### ISSUE THREAD

**Issue by mammuth** [WhisperSystems/TextSecure#4082]
It seems like TextSecure does not ask for "new" messages when regaining internet connection after being in airplane mode or something similar. Is this assumption correct?

Reproduce:
- Switch on airplane mode
- Make someone write you a message
- Switch off airplane mode

--> Notification for the new message won't get triggered
- Open TextSecure app

--> New message is displayed

Expected:
- It should listen to the system event of regaining an internet connection and then automatically download new messages and notify about them.

**Comment by moxie0** [comment]
If that's perfectly reproducible, can you post a debug log of the entire sequence?  The GCM message should come in a few moments after regaining network access.

**Comment by mammuth** [comment]
Hm, seems like I'm not able to reproduce it now... I'm going to close this issue until I've found more details and a logcat.

**Comment by McLoo** [comment]
Seems to hit my experience. Though even not completely reproducible.
Often in airplane and in bad mobile signal area. Hasn't @2-4601 reported the delay with something about a bad signal in the basement ?

**Comment by 2-4601** [comment]
Yeah, in #3925 the latter two cases are about receiving messages only after coming out of poor reception area AND opening TS. But the first two cases in the same issue happened in a stable Wi-Fi environment. I guess there might have been a temporary glitch in the Wi-Fi connection but so far I have never noticed any connection issues with other devices or software.

If someone feels up for it, try to put one device in a fridge or in a pot and send a TS message to it from another device. Unfortunately I don't have time to test this now.

**Comment by mammuth** [comment]
Today I experienced a similar problem

Since I wasn't able to reproduce it, I'm not filing a new issue.

1. I sent a message to a friend
2. We waited half a minute but it didn't appear
3. Sent a test message via facebook messenger for testing whether the receiving phone has a internet or gcm problem
4. facebook message instantly appeared on the receiving phone
5. Sent a second message via TextSecure to the recieving phone
6. Didn't notify either
7. Open TextSecure, messages appear after some seconds
8. filed this bug report, which doesn't look that informative to me: https://gist.github.com/anonymous/013538c4d57ec73d0c44
9. Swiped away TS from recent apps
10. Sent a test message again via TS
11. Message appears instantly this time


Some infos: 
The receiving device has a (slow) WiFi connection. A VPN connection via Cisco Anyconnect is established. It's running Stock Android 5.1.1
